pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+                           'repository %s' % (cname, ke))
+
+    ##########################################################
+    #
+    #              Faked Qualifier methods
+    #
+    ###########################################################
+
+    def _fake_enumeratequalifiers(self, namespace, **params):
+        # pylint: disable=unused-argument
+        """
+        Imlements a mock server reactor for
+        :meth:`~pywbem.WBEMConnection.EnumerateQualifiers`
+
+        Enumerates the qualifier declarations in the local repository of this
+        namespace.
+
+        Returns: TODO
+
+        Exceptions:
+            CIMError
+
+        """
+        try:
+            qualifiers = list(self.qualifiers[namespace].values())
+        except KeyError:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
+        return self._make_tuple(qualifiers)
+
+    def _fake_getqualifier(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`pywbem.WBEMConnection.GetQualifier`.
+
+        Retrieves a qualifier declaration from the local repository of this
+        namespace.
+
+        Return:
+          Returns a tuple representing the _imethodcall return for this
+          method where the data is a QualifierDeclaration
+
+        Exceptions:
+            CIMError CIM_ERR_INVALID_NAMESPACE, CIMError(CIM_ERR_NOT_FOUND
+        """
+        if not self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
+
+        qname = params['QualifierName']
+
+        try:
+            qualifier = self.qualifiers[namespace][qname]
+            return self._make_tuple([qualifier])
+        except KeyError:
+            ce = CIMError(CIM_ERR_NOT_FOUND, qname)
+            raise ce
+
+    def _fake_setqualifier(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`pywbem.WBEMConnection.SetQualifier`.
+
+        Create or modify a qualifier type in the local repository of this
+        class.  This method will create a new namespace for the qualifier
+        if none is defined.
+
+        Return: TODO
+
+        Exceptions:
+            CIMError CIM_ERR_INVALID_PARAMETER or
+            CIMError(CIM_ERR_ALREADY_EXISTS
+        """
+        qual = params['QualifierDeclaration']
+
+        if namespace not in self.qualifiers:
+            self.qualifiers[namespace] = NocaseDict({})
+
+        if not isinstance(qual, CIMQualifierDeclaration):
+            raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                           'The qualifier declaration parameter is not a '
+                           'valid CIMQualifierDeclaration: %s' % qual)
+
+        if qual.name in self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_ALREADY_EXISTS, '%s' % qual.name)
```

Comment:
We should have a human readable message in the second argument, not just the qualifier name.

Reply:
Done